OSDOCS-11158: adds image mode to MicroShift install (#80599)
openshift/openshift-docs: modules/microshift-install-rhel-image-mode-run-container.adoc

.Prerequisites

* A {op-system-base-full} 9.4 host with an active Red Hat subscription for building {microshift-short} bootc images and running containers.
* You are logged into the {op-system} 9.4 host using the user credentials that have `sudo` permissions.
* You have a pull secret file for downloading the required {microshift-short} container images.

.Procedure

. Run the container for the {microshift-short} service by entering the following command:
+
[source,terminal]
----
PULL_SECRET=~/.pull-secret.json
IMAGE_NAME=microshift-4.16-bootc

$ sudo podman run --rm -it --privileged \
    -v "${PULL_SECRET}":/etc/crio/openshift-pull-secret:ro \
    -v /var/lib/containers/storage:/var/lib/containers/storage \
    --name "${IMAGE_NAME}" \
    "${IMAGE_NAME}"
----
+
[NOTE]
====
The `systemd-modules-load` service fails to start in the container if the host kernel version is different from the bootc image kernel version. This failure can be safely ignored as all the necessary kernel modules have been loaded by the host.
====

. A login prompt is presented in the terminal after {microshift-short} has started.

> . Log into the running container using the your user credentials.

Reviewer: ```suggestion
. Log in to the running container using the your user credentials.
```

[source,terminal]
----
NAMESPACE                  NAME                                       READY   STATUS    RESTARTS      AGE
kube-system                csi-snapshot-controller-7cfb9df49c-kc9dx   1/1     Running   0             31s
kube-system                csi-snapshot-webhook-5c6b978878-jzk5r      1/1     Running   0             28s
openshift-dns              dns-default-rpnlt                          2/2     Running   0             14s
openshift-dns              node-resolver-rxvdk                        1/1     Running   0             31s
openshift-ingress          router-default-69cd7b5545-7zcw7            1/1     Running   0             29s
openshift-ovn-kubernetes   ovnkube-master-c7hlh                       4/4     Running   1 (16s ago)   31s
openshift-ovn-kubernetes   ovnkube-node-mkpht                         1/1     Running   1 (17s ago)   31s
openshift-service-ca       service-ca-5d5d96459d-5pd5s                1/1     Running   0             28s
openshift-storage          topolvm-controller-677cbfcdb9-28dqr        5/5     Running   0             31s
openshift-storage          topolvm-node-6fzbl                         3/3     Running   0             14s
----

> . Now you can use your {microshift-short} running in the container like any {microshift-short} cluster.

Reviewer: ```suggestion
. Now you can use your {microshift-short} running in the container like any {microshift-short} cluster.
```
The IBM style mentions to use "such as" as opposed to like, but I don't think that makes sense here. Just thought I'd note it just in case.

// Module included in the following assemblies:
//
// microshift_install_bootc/microshift-install-rhel-image-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-rhel-image-mode-run-container_{context}"]
= Running the {microshift-short} bootc container

> If the host is already configured to have a `rhel` volume group with free space, this configuration is inherited by the container. It can be used by the {microshift-short} LVMS Container Storage Interface (CSI) to allocate storage.

Reviewer 2: I do not think this section belongs here. We've already described the CSI options in the previous one.

.Prerequisites

* A {op-system-base-full} 9.4 host with an active Red Hat subscription for building {microshift-short} bootc images and running containers.

> * You are logged into the {op-system} 9.4 host using the user credentials that have SUDO permissions configured.

Reviewer 3: ```suggestion
* You are logged into the {op-system} 9.4 host using the user credentials that have oot access permissions configured.
```

* You have a pull secret file for downloading the required {microshift-short} container images.

> * You configured host container storage for reusing available container images.

Reviewer 2: It's not necessary.
```suggestion
```

> * You configure host container storage for reusing available container images.

Reviewer 3: ```suggestion
* You configured a host container storage for reusing available container images.
```

Reviewer 3: Not sure if it should be "configure"  here or not.

Author: Yep, should be past tense, we did that already.

.Procedure

. Run the container for the {microshift-short} service by entering the following command:
+
[source,terminal]
----
PULL_SECRET=~/.pull-secret.json
IMAGE_NAME=microshift-4.16-bootc

$ sudo podman run --rm -it --privileged \
    -v "${PULL_SECRET}":/etc/crio/openshift-pull-secret:ro \
    -v /var/lib/containers/storage:/var/lib/containers/storage \
    --name "${IMAGE_NAME}" \
    "${IMAGE_NAME}"
----
//moved the hotwiring openvswitch over to CNI module

[NOTE]
====
The `systemd-modules-load` service fails to start in the container if the host kernel version is different from the bootc image kernel version. This failure can be safely ignored as all the necessary kernel modules have been loaded by the host.
====

A login prompt like the following is presented in the terminal after {microshift-short} has started:

> +

Reviewer 2: Extra +
```suggestion
```

[source,terminal]
----

> $ tbd example

Reviewer 2: Well, it may look like this, but I'm not sure we should be displaying it.

```
Red Hat Enterprise Linux 9.20240820.0.4 (Plow)
Kernel 5.14.0-427.28.1.el9_4.x86_64 on an x86_64

5b2a7019fdf9 login: 
```

Author: That's fair. Removed that and just stated what should happen as part of the next step.

----

. Log into the running container using the `redhat:<password>` credentials.
+
[source,terminal]
----
$ tbd example

> ----

Reviewer 2: No need for an example. It's just entering the user and password

. Verify that all the {microshift-short} pods are up and running without errors by running the following command:
+
[source,terminal]
----
$ watch sudo oc get pods -A \
    --kubeconfig /var/lib/microshift/resources/kubeadmin/kubeconfig
----
.Example output
[source,text]
----

> Run the sudo shutdown now command to stop the container.

Reviewer 2: The `sudo shutdown now` is a command to be run within the container to shut it down. 
If we mention this, we should do it in the end and as a separate section.

Author: Oh, I see. Do we have an output from a cluster? Does it look like any other MicroShift example? I added the vanilla content from our usual snippet. WDYT?

> .Example output
> [source,text]
> ----
> Run the sudo shutdown now command to stop the container.
> ----

Reviewer 2: ```suggestion
.Example output
[source,text]
----
NAMESPACE                  NAME                                       READY   STATUS    RESTARTS      AGE
kube-system                csi-snapshot-controller-7cfb9df49c-kc9dx   1/1     Running   0             31s
kube-system                csi-snapshot-webhook-5c6b978878-jzk5r      1/1     Running   0             28s
openshift-dns              dns-default-rpnlt                          2/2     Running   0             14s
openshift-dns              node-resolver-rxvdk                        1/1     Running   0             31s
openshift-ingress          router-default-69cd7b5545-7zcw7            1/1     Running   0             29s
openshift-ovn-kubernetes   ovnkube-master-c7hlh                       4/4     Running   1 (16s ago)   31s
openshift-ovn-kubernetes   ovnkube-node-mkpht                         1/1     Running   1 (17s ago)   31s
openshift-service-ca       service-ca-5d5d96459d-5pd5s                1/1     Running   0             28s
openshift-storage          topolvm-controller-677cbfcdb9-28dqr        5/5     Running   0             31s
openshift-storage          topolvm-node-6fzbl                         3/3     Running   0             14s
----
```

//is that it?

> //now what? do we deploy this somewhere to test it, do more config?

Reviewer 2: Now, there's a running MicroShift inside the container, so users would use it like any other MicroShift cluster instance.
